### ISSUE TITLE
`selector-no-redundant-nesting-selector` false positive when using multiple `&`s

### ISSUE THREAD

**Issue by FloEdelmann** [stylelint-scss/stylelint-scss#598]
The `selector-no-redundant-nesting-selector` rule warns about this pattern, although it cannot be avoided:

```scss
.foo {
    & > .bar:not(&--foo):not(&--bar):not(&--baz) {
//  ^ Unnecessary nesting selector (&) (scss/selector-no-redundant-nesting-selector)
        background: red;
    }
}

// compiles to:
.foo > .bar:not(.foo--foo):not(.foo--bar):not(.foo--baz) {
    background: red;
}
```

When rewriting to the following (as suggested by the rule), the compiled result becomes invalid:

```scss
.foo {
    > .bar:not(&--foo):not(&--bar):not(&--baz) {
        background: red;
    }
}

// compiles to:
// (note missing `.foo` at the start)
> .bar:not(.foo--foo):not(.foo--bar):not(.foo--baz) {
    background: red;
}
```

This was already reported in #472 for Less, but the corresponding PR didn't fix this particular issue.

**Versions:**
node-sass 4.14.1
stylelint 14.1.0
stylelint-scss 4.0.0

**Comment by kristerkari** [comment]
Thanks! Looks like a bug.
